src/bdecode.cpp

```diff
@@ -477,22 +477,22 @@ namespace libtorrent
 
 		while (tokens[token].type != bdecode_token::end)
 		{
-			bdecode_token const& t = tokens[token];
-			TORRENT_ASSERT(t.type == bdecode_token::string);
-			int const size = m_root_tokens[token + 1].offset - t.offset - t.start_offset();
+			bdecode_token const* t = &tokens[token];
+			TORRENT_ASSERT(t->type == bdecode_token::string);
+			int const size = (t + 1)->offset - t->offset - t->start_offset();
```

Comment:
it's not obvious from just looking at the code here that m_root_tokens[token + 1] is the same as (t + 1). If it is, it should also be the same as ``tokens[token+1]``, I would think compilers would have an easier time to optimize that